lib/chef/resource/sudo.rb

```diff
@@ -167,6 +167,7 @@ def platform_config_prefix
             mode "0440"
             variables new_resource.variables
             verify "#{new_resource.visudo_binary} -cf %{path}" if visudo_present?
```

Comment:
I think we can actually simplify this a bit:
```ruby
verify "#{new_resource.visudo_binary} -cf #{new_resource.config_prefix}/sudoers" if visudo_present?
```

sudoers will parse in all the .d files so there's no need to parse both the sudoers config and the .d file we're working on now:

```
$ visudo -cf /etc/sudoers
/etc/sudoers: parsed OK
/etc/sudoers.d/90-cloud-init-users: parsed OK
/etc/sudoers.d/README: parsed OK```

Reply:
you have to combined the sudoers.d with the %{path} file though not the old /etc/sudoers file.

either way though that line should get replaced.

Reply:
I have removed `verify "#{new_resource.visudo_binary} -cf %{path}" if visudo_present?`, because if newly added sudo resource is invalid then the overall state will be invalid.